### PR TITLE
feat/tr 4060/partially answered items non-default config

### PR DESCRIPTION
**Pull request by oatymart**
https://oat-sa.atlassian.net/browse/TR-4060

**Context**: in test-taker reviews, the test should be able to display an item as **unanswered** if it was partially answered. This goes against the default QTI behaviour, so it should depend on a platform configuration.

Required by:
- https://github.com/oat-sa/nfer-enot-template/pull/55

Required by & requires:
- [x] https://github.com/oat-sa/tao-test-runner-qti-fe/pull/486

The various places ([1](https://github.com/oat-sa/qti-sdk/blob/develop/src/qtism/runtime/tests/AssessmentItemSession.php#L1109), [2](https://github.com/oat-sa/extension-tao-testqti/blob/master/helpers/class.TestRunnerUtils.php#L744), [3](https://github.com/oat-sa/extension-tao-testqti/blob/master/models/classes/runner/QtiRunnerService.php#L1335)) where item answeredness is computed were already set up to take a `partially` param, but nobody ever passed `false` before.

New config value `review.partiallyAnsweredIsAnswered` is read and applied inside:
- `QtiRunnerService->getTestContext()`
- `QtiRunnerService->getItemSession()`
- `QtiRunnerMap->getScopedMap()`
before reaching the above-linked places.

(Commit https://github.com/oat-sa/extension-tao-testqti/pull/2260/commits/9afaf63e692448081329c0ebbd705eaec85115b1 is testable on http://test-tr-4046.playground.kitchen.it.taocloud.org:49751/)

### CODE REVIEW

**Review by jsconan — COMMENTED** [review]
It looks ok, except the new option will only be set upon install, the migration script is missing.

**Comment by oatymart** [comment]
> It looks ok, except the new option will only be set upon install, the migration script is missing.

I thought it would not be necessary, since the absence of the setting is interpreted by the code the same way as having the default setting. But, it probably looks better to add it, so I may do that.

File: models/classes/runner/QtiRunnerService.php [hunk 1]
```diff
@@ -1073,7 +1074,10 @@ public function getItemSession(RunnerServiceContext $context)
         // The current item answered state
         $route    = $session->getRoute();
         $position = $route->getPosition();
-        $output['itemAnswered'] = TestRunnerUtils::isItemCompleted($route->getRouteItemAt($position), $itemSession);
+        $config = $this->getTestConfig();
+        $reviewConfig = $config->getConfigValue('review');
+        $partiallyAnsweredIsAnswered = isset($reviewConfig['partiallyAnsweredIsAnswered']) ? (bool) $reviewConfig['partiallyAnsweredIsAnswered'] : true;
```
Comment (olga-kulish) [inline models/classes/runner/QtiRunnerService.php:1079]: Is it possible for `$reviewConfig` to be undefined/null? If yes, should check for it.

**Review by jsconan — APPROVED** [review]
**Review Checklist**
- [ ] ~~New code is covered by tests (if applicable)~~
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] ~~New code is not subject to concurrency issues (if applicable)~~
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic

File: models/classes/runner/QtiRunnerService.php (hunk 1 above)
Comment (olga-kulish) [reply models/classes/runner/QtiRunnerService.php:1079]: I see it's not checked in [other places](https://github.com/oat-sa/extension-tao-testqti/blob/develop/models/classes/runner/map/QtiRunnerMap.php#L159),  so ignore it

**Review by olga-kulish — APPROVED** [review]
Good job!

**Review by wazelin — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Pull request merged**